client/transaction-pool/src/lib.rs

```diff
@@ -348,6 +418,9 @@ impl<PoolApi, Block> MaintainedTransactionPool for BasicPool<PoolApi, Block>
 						}
 					}
 
+					let extra_pool = pool.clone();
+					ready_poll.lock().trigger(block_number, move || Box::new(extra_pool.validated_pool().ready()));
```

Comment:
As mentioned in private convos, the `ChainEvent::NewEvent` is not super reliable, so we might end up skipping block `trigger`. We should switch to `all_blocks_import_notification_stream` when #5200 lands

Reply:
I am not sure, but we'll see, I'll add a reference/todo

Not sure because maybe `ChainEvent::NewBlock` is "unreliable" in a sense that we need here - so if we are catching up with network, we also shouldn't make attempts to produce blocks